src/Symfony/Component/Form/ChoiceList/Factory/Cache/ChoiceLoader.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\ChoiceList\Factory\Cache;
+
+use Symfony\Component\Form\ChoiceList\Factory\CachingFactoryDecorator;
+use Symfony\Component\Form\ChoiceList\Loader\ChoiceLoaderInterface;
+use Symfony\Component\Form\FormTypeInterface;
+
+/**
+ * A FlyWeight for types using static loaders, used by {@see CachingFactoryDecorator}.
+ *
+ * @internal
+ *
+ * @author Jules Pietri <jules@heahprod.com>
+ */
+class ChoiceLoader implements ChoiceLoaderInterface
+{
+    private static $loaders = [];
+
+    /** @var ChoiceLoaderInterface */
+    private $loader;
+
+    /**
+     * @param FormTypeInterface     $formType A form type configuring a list
+     * @param ChoiceLoaderInterface $loader   A static loader used by the type
+     * @param mixed                 $vary     Dynamic data used to compute a unique hash when caching the loader
+     */
+    final public function __construct(FormTypeInterface $formType, ChoiceLoaderInterface $loader, $vary)
```

Comment:
Why `$vary` is optional in `AbstractStaticCallback` and required here?
```suggestion
    final public function __construct(FormTypeInterface $formType, ChoiceLoaderInterface $loader, $vary = null)
```

Reply:
A mistake, thanks!